buildSrc/src/main/java/org/eclipse/dataspaceconnector/gradle/ModuleNamesPlugin.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.gradle;
+
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toMap;
+
+/**
+ * Custom grade plugin to avoid module name duplications.
+ * Checks between modules with a gradle build file that their names are unique in the whole project.
+ * `samples` and `system-tests` modules are excluded
```

Comment:
Consider adding a reference to the gradle issue that is making this rule necessary in the comment.